pkg/sql/drop_index.go

```diff
@@ -146,6 +146,7 @@ func (n *dropIndexNode) startExec(params runParams) error {
 		// in the primary index. Dropping these columns will lead to data loss.
 		if idx != nil {
 			primaryStoredColumns := tableDesc.GetPrimaryIndex().CollectPrimaryStoredColumnIDs()
+			tableDesc.GetPrimaryIndex().CollectKeyColumnIDs().ForEach(primaryStoredColumns.Add)
```

Comment:
nit: would this be more clear now if we rename `primaryStoredColumns` to just `primaryIdxColumns`? it looks like now it will have both key and stored columns.